app/src/main/java/kr/ac/kaist/team888/util/FeatureController.java

```diff
@@ -0,0 +1,73 @@
+package kr.ac.kaist.team888.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+public class FeatureController {
+  
+  /**
+   * Interface for listening change of hangul features.
+   *
+   * <p> If some feature(weight, curve, and etc)'s value are change,
+   * {@link FeatureController} calls {@link #onFeatureChange()}.
+   *
+   */
+  public interface OnFeatureChangeListener {
+    /**
+     * Method that will be executed on feature's value change.
+     */
+    void onFeatureChange();
+
+    /**
+     * Priority of listener.
+     *
+     * <p> The lower priority executed first.
+     *
+     * @return priority as integer
+     */
+    int getPriority();
+  }
+
+  private static class Singleton {
+    private static final FeatureController instance = new FeatureController();
+  }
+
+  /**
+   * Getter of singleton instance.
+   *
+   * @return singleton instance
+   */
+  public static FeatureController getInstance() {
+    return FeatureController.Singleton.instance;
+  }
+
+  private ArrayList<OnFeatureChangeListener> listeners;
```

Comment:
Is there any situation to need to iterate? `listeners.poll()` returns the element of highest priority automatically. Doesn't need to iterate, doesn't need to sort.